charts/nri-metadata-injection/templates/admission-webhooks/job-patch/psp.yaml

```diff
@@ -1,4 +1,4 @@
-{{- if (and (not .Values.customTLSCertificate) (not .Values.certManager.enabled)) }}
+{{- if (and (not .Values.customTLSCertificate) (not .Values.certManager.enabled) (.Values.rbac.pspEnabled)) }}
```

Comment:
would make sense to remove them automatically for 1.25 only instead of having a parameter?

Reply:
Something like this exist?
```
{{- if semverCompare ">=1.19-0" .Capabilities.KubeVersion.GitVersion -}}
apiVersion: networking.k8s.io/v1
{{- else if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion -}}
apiVersion: networking.k8s.io/v1beta1
{{- else -}}
apiVersion: extensions/v1beta1
{{- end }}
```

https://github.com/helm/helm/blob/d4cc130fa90027e8c1c69ad2a06c6dee4c31d29d/pkg/chartutil/create.go#L223

Reply:
![Uploading Screenshot 2022-10-21 at 11.18.17.png…]()

Reply:
```
Capabilities: This provides information about what capabilities the Kubernetes cluster supports.
      Capabilities.APIVersions is a set of versions.
      Capabilities.APIVersions.Has $version indicates whether a version (e.g., batch/v1) or resource (e.g., apps/v1/Deployment) is available on the cluster.
      Capabilities.KubeVersion and Capabilities.KubeVersion.Version is the Kubernetes version.
      Capabilities.KubeVersion.Major is the Kubernetes major version.
      Capabilities.KubeVersion.Minor is the Kubernetes minor version.
      Capabilities.HelmVersion is the object containing the Helm Version details, it is the same output of helm version
      Capabilities.HelmVersion.Version is the current Helm version in semver format.
      Capabilities.HelmVersion.GitCommit is the Helm git sha1.
      Capabilities.HelmVersion.GitTreeState is the state of the Helm git tree.
      Capabilities.HelmVersion.GoVersion is the version of the Go compiler used.
```

Reply:
It exists, but won't be taken into account if you run (for example) `helm template`.

PSPs have always had controversy and limited adoption, and are deprecated in around half of the version bracket we support. I don't see a strong reason why we should not make them opt-in for earlier versions too.

Reply:
oh, the manifest generator 😢